### ISSUE TITLE
prov/tcp: stacking “struct ofi_epollfds_event events[MAX_POLL_EVENTS];" in tcpx_conn_mgr_run() is too much

### ISSUE THREAD

**Issue by bfaccini** [ofiwg/libfabric#8696]
**Describe the bug**
Starting to run Libfabric 1.17 with DAOS (also using Argobots with a default ULT stack size of 16K...), we have experienced multiple SEGVs or assert/abort.
Corefiles analysis has leaded to a ULT stack overflow, where tcpx_conn_mgr_run() was involved. This is where I have found that since commit 58629529eb0fa83c72cafde2346e74dbc1f5b42c, MAX_POLL_EVENTS has been bumped from 100 to 1024, causing events[] array to consume about 28K in stack !!

**To Reproduce**

**Expected behavior**
events[] array must be allocated out of stack, or at least a minimal stack size must be indicated when running with TCP provider.

**Output**
If applicable, add output to help explain your problem. (e.g. backtrace, debug logs)

**Environment:**
OS (if not Linux), provider, endpoint type, etc.

**Additional context**

**Comment by soumagne** [comment]
@shefty it sounds to me like it should not only be reduced but it should first be dynamically allocated using a reasonable size and then it should grow / be realloc'ed / doubled depending on whether or not the number events returned by epoll_wait() reaches the max number of events that the array can hold ? Something like:
```C
nfds = epoll_wait(..., events, max_events,...);
...
if (nfds == max_events && max_events * 2 <= MAX_EVENTS) {
        events = realloc(events, sizeof(*events) * max_events * 2);
        max_events *= 2;
}
```

**Comment by frostedcmos** [comment]
In order to unblock us for updating to 1.17.0+ ofi in DAOS we are going to be applying local patch for now to unroll number of events back to 100 (https://daosio.atlassian.net/browse/DAOS-12407)

**Comment by shefty** [comment]
That sounds reasonable.  We'll have a fix applied prior to the 1.18 release and will see how to update 1.17.

**Comment by shefty** [comment]
Fixed in v1.17.x branch.  See #8725 for an update for v1.18.x

**Comment by shefty** [comment]
Update was also made for io_uring.
